provisioning/roles/php/tasks/debian-repo.yml

```diff
@@ -2,23 +2,27 @@
   fail: msg="Jessie only supports version 5.6."
   when: "ansible_lsb.major_release|int >= 8 and {{ php_version_installed }} < 5.6"
 
-- name: debian-repo | Install DotDeb key
+- name: debian-repo | error when trying to use something less than PHP 7 on Stretch
+  fail: msg="Stretch only supports version 7"
+  when: "ansible_lsb.major_release|int >= 9 and {{ php_version_installed }} < 7.0"
+
+- name: debian-repo | Install DotDeb key on Jessie
   apt_key: url=https://www.dotdeb.org/dotdeb.gpg
   become: yes
-  when: "{{ php_version_installed }} > 5.4"
+  when: "ansible_lsb.major_release|int <= 8 and {{ php_version_installed }} > 5.4"
```

Comment:
I know this was already wrong, but could you use `|version_compare('5.4', '>')` instead, just like it's done below?